api/gateway-operator/v2alpha1/controlplane_types.go

```diff
@@ -457,3 +462,81 @@ func (c *ControlPlane) SetConditions(conditions []metav1.Condition) {
 func (c *ControlPlane) GetExtensions() []commonv1alpha1.ExtensionRef {
 	return c.Spec.Extensions
 }
+
+// ControlPlaneKonnectOptions defines the Konnect-related configuration options for the ControlPlane.
+//
+// +apireference:kgo:include
+type ControlPlaneKonnectOptions struct {
+	// ConsumersSync indicates whether consumer synchronization with Konnect is enabled.
+	//
+	// +optional
+	// +kubebuilder:default=enabled
+	// +kubebuilder:validation:Enum=enabled;disabled
+	ConsumersSync *ControlPlaneKonnectConsumersSyncState `json:"consumersSync,omitempty"`
+
+	// Licensing defines the configuration for Konnect licensing.
+	//
+	// +optional
+	Licensing *ControlPlaneKonnectLicensing `json:"licensing,omitempty"`
+
+	// NodeRefreshPeriod is the period for refreshing the node information in Konnect.
+	//
+	// +optional
+	NodeRefreshPeriod *metav1.Duration `json:"nodeRefreshPeriod,omitempty"`
+
+	// ConfigUploadPeriod is the period for uploading configuration to Konnect.
+	//
+	// +optional
+	ConfigUploadPeriod *metav1.Duration `json:"configUploadPeriod,omitempty"`
+}
+
+// ControlPlaneKonnectConsumersSyncState defines the state of consumer synchronization with Konnect.
+type ControlPlaneKonnectConsumersSyncState string
+
+const (
+	// ControlPlaneKonnectConsumersSyncStateEnabled indicates that consumer synchronization is enabled.
+	ControlPlaneKonnectConsumersSyncStateEnabled ControlPlaneKonnectConsumersSyncState = "enabled"
+	// ControlPlaneKonnectConsumersSyncStateDisabled indicates that consumer synchronization is disabled.
+	ControlPlaneKonnectConsumersSyncStateDisabled ControlPlaneKonnectConsumersSyncState = "disabled"
+)
+
+// ControlPlaneKonnectLicensing defines the configuration for Konnect licensing.
+//
+// +apireference:kgo:include
+type ControlPlaneKonnectLicensing struct {
+	// Enabled indicates whether Konnect licensing is enabled.
+	//
+	// +optional
+	// +kubebuilder:default=disabled
+	// +kubebuilder:validation:Enum=enabled;disabled
+	Enabled *ControlPlaneKonnectLicensingState `json:"enabled,omitempty"`
+
+	// InitialPollingPeriod is the initial polling period for license checks.
+	//
+	// +optional
+	InitialPollingPeriod *metav1.Duration `json:"initialPollingPeriod,omitempty"`
+
+	// PollingPeriod is the polling period for license checks.
+	//
+	// +optional
+	PollingPeriod *metav1.Duration `json:"pollingPeriod,omitempty"`
```

Comment:
Shouldn't we add CEL rules to require these to be set when Konnect license functionality is enabled?

Reply:
Added here https://github.com/Kong/kubernetes-configuration/pull/535/commits/3b7bf3db2fc31d50cd8b018edc02982a788e616c

Reply:
However, for the current rules I added, I think they may need to be modified. These two fields have default values, so they are not necessary. What do you think?

Reply:
Changed. https://github.com/Kong/kubernetes-configuration/pull/535/commits/665ef8ba43e570c01423039fe8c21c6ae31a7982